src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4584,21 +4623,26 @@ private SeparatedSyntaxList<ParameterSyntax> ParseParameterList(
             out SyntaxToken open,
             out SyntaxToken close,
             SyntaxKind openKind,
-            SyntaxKind closeKind)
+            SyntaxKind closeKind,
+            bool forExtension)
         {
             open = this.EatToken(openKind);
 
             var saveTerm = _termState;
             _termState |= TerminatorState.IsEndOfParameterList;
 
+            Func<LanguageParser, ParameterSyntax> parseElement = forExtension
+                    ? static @this => @this.ParseParameter(allowOptionalIdentifier: true)
+                    : static @this => @this.ParseParameter(allowOptionalIdentifier: false);
+
             var parameters = ParseCommaSeparatedSyntaxList(
                 ref open,
                 closeKind,
                 static @this => @this.IsPossibleParameter(),
-                static @this => @this.ParseParameter(),
+                parseElement,
                 skipBadParameterListTokens,
                 allowTrailingSeparator: false,
-                requireOneElement: false,
+                requireOneElement: forExtension, // For extension declarations, we require one receiver parameter
```

Comment:
>requireOneElement: forExtension, // For extension declarations, we require one receiver parameter [](http://example.com/codeflow?start=16&length=97)

It looks like the meaning of this parameter is to ensure that the syntax list is not empty, i.e. at least one item is expected. That doesn't mean that the helper won't accept more that one item. The comment, however, might be interpreted as though we instruct the helper to accept exactly one item (no more, no less).